File: jinxedjewelry/submissions/accepted/jannik.java
import java.util.*;

public class jannik {
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		for (int t = in.nextInt(); t-->0; ) {
			int n = in.nextInt();
			long[] a = new long[n];
			for (int i = 0; i < n; i++) a[i] = in.nextLong();
			Arrays.sort(a);

			int front = 0, back = n-1, res = 1;
			for ( ; back - front + 1 >= 3; res++) {
				back--;
				a[back] += a[back+1] + 1; // this is irrelevant
				if (--a[front] == 0) front++;
			}
			if (back - front + 1 == 2 && a[front] > 1)
				res++;
			System.out.println(res);
		}
	}
}

File: jinxedjewelry/submissions/accepted/jannik.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <queue>

using namespace std;

using ll = long long;
using vll = vector<ll>;

int main() {
	ll t; cin >> t;
	while (t-->0) {
		ll n; cin >> n;
		deque<ll> a;
		for (ll i = 0; i < n; i++) {
			ll x; cin >> x;
			a.emplace_back(x);
		}
		sort(a.begin(), a.end());
		ll res = 0;
		for (; a.size() >= 3; res++) {
			ll f = a.front(); a.pop_front();
			ll new_c = a.back(); a.pop_back();
			new_c += a.back(); a.pop_back();
			new_c++, f--;
			a.emplace_back(new_c);
			if (f > 0) a.emplace_front(f);
		}
		if (a.size() == 2)
			if (a.front() > 1) res++; // merge the two chains
		res++; // join single remaining chain
		cout << res << endl;
	}
	return 0;
}
